### ISSUE TITLE
Difference in sfx noise behaviour switching from 0.70.6 to 0.80-dev

### ISSUE THREAD

**Issue by nalquas** [nesbox/TIC-80#1140]
There appears to be some sort of change in how sfx is handling noise playback.
First noticed this in my game [Mainfighter](https://tic.computer/play?cart=287) when trying it in 0.80-dev, here's how to reproduce it:
```lua
-- title:  Noise test
-- author: Nalquas
-- desc:   Attempting to reproduce a bug I observed in Mainfighter
-- script: lua

function TIC()
	cls(0)
	if btn(4) then
		sfx(0,"C#4",-1,0,8,0)
		print("Noise C#4")
	else
		sfx(0,"C-4",-1,0,8,0)
		print("Noise C-4")
	end
end
```
The sound effect 0 looks as follows:
![Screenshot_20200727_095247](https://user-images.githubusercontent.com/22664233/88517338-fb175d80-cfee-11ea-81e9-371381eafae2.png)
And finally, here's the cartridge (GitHub doesn't let me upload it directly, so here's a zipped version): [noisetest.tic.zip](https://github.com/nesbox/TIC-80/files/4980673/noisetest.tic.zip)

Here's how the difference actually sounds (uploaded to Youtube because GitHub doesn't allow embedding of videos...):
First, TIC-80 0.70.6:
https://www.youtube.com/watch?v=H3-I2KurdR0
And TIC-80 0.80.1267-dev Pro (built from source through AUR, Linux 64bit):
https://www.youtube.com/watch?v=o4cCFK0ySOI

**Comment by nesbox** [comment]
Thank you for the detailed report, we will investigate this.

**Comment by rahul625** [comment]
Looks like we won't get 0.80 this year ;-;

**Comment by nalquas** [comment]
To help narrow the issue down, I compiled some old commits to see where noise broke.
The first broken commit appears to be 8b158b073e42bfae0460cd1c9d88f1c8b0a8c9af, so the issue has to be in this commit somewhere.
For the record, here are all the commits I've tested:
```
WORKIN e7f4746b3c23e6e9b980fc1f02eb5be29fca5c34 Mar 15
WORKIN 383aabf23ff905a37a1bd946c1d00a52d8cda229 Apr 17
WORKIN dc817a127d74852ede98f631fb1987a05a7ba0ca Apr 20
WORKIN a2015f4716e5c12df9a16c42377973b5ddfaa9c6 Apr 28
WORKIN 0d7ef8439c0dc45821fbdd29c1b51a25b7126a62 Apr 30
WORKIN 7043d2b94228876daeee582652ad00a8992f0f17 May 01 -- LAST WORKING COMMIT
BROKEN 8b158b073e42bfae0460cd1c9d88f1c8b0a8c9af May 02 -- FIRST BROKEN COMMIT
BROKEN 0717a19bd17e6cc6eca43dbfbe694ddda3a0f8a5 May 03
BROKEN 49547daa451043aa3c4da09dab7ead6d7fdcb887 May 17
```

**Comment by nesbox** [comment]
Fixed here 5cd8e42, pls check it
Thank you.

**Comment by nalquas** [comment]
Yup, works perfectly now.
Thank you for the fix.
